### ISSUE TITLE
Resample filtered data + Mask + download

### ISSUE THREAD

**Issue by soazig** [berkeley-stat159/project-epsilon#209]
Hello @jarrodmillman  @matthew-brett @rossbar 

We have been exploring the filtered data for comparison across subjects.

Thank you for the provided template for the filtered data : https://nipy.bic.berkeley.edu/rcsds/mni_icbm152_nlin_asym_09c_2mm/resample_templates.py

In the script, it seems like they apply all of the mni_icbm\* files on the data.
Do we have to do that or can we just use the 2 most interesting you mentioned : 
to display pattern: mni_icbm152_t1_tal_nlin_asym_09c_2mm.nii
and the mask : mni_icbm152_t1_tal_nlin_asym_09c_mask_2mm.nii
My plan is to apply the mask first and then the template to visualize the data.
=> Do you think that is sufficient ?

For unfiltered data, I was using the technique shown in class to come up with a mask (with the histogram plots). Other groups are using a file in the anatomy folder of the data called 'inplane_brain.nii.gz", Our group have the data called "highres001_brain_mask.nii.gz".
The orther files are highres001.nii.gz, highres001_brain.nii.gz, inplane.nii.gz.
=> Do you think we can use one of this file to mask the data for each subject's run's?
=> I think, *_brain.nii.gz can be used to select the voxels in the brain. I am not sure about the difference between 'inplane' and 'highresr001', or even what the highres001_brain_mask.nii.gz represents.

Also, we have a discussion in our group on whether programming the download of the .zip files for the filtered data or download them one by one.
I checked the hashes for 2 files downloaded in the 2 different ways and they are the same.
We are just a bit suspicious because downloading the files take ~10s by file while downloading the .zip takes 5hours.
=> Do you have any thoughts on that ?

Thank you !
@mingujo @ye-zhi @lizhua @timothy1191xa

**Comment by matthew-brett** [comment]
Sorry - the `resample_templates.py` is just to show you how I generated the files in that directory from the originals.  You don't need to use that script.  So yes, I suggest just applying the mask and visualizing with the T1 image.

If you are working on the preprocessed data, I would use the template mask as above.   If you are using the raw data, it's perfectly reasonable to use the technique I was showing in class, eyeballing the histogram.  If you want to use subject's `highres` image mask, in `highres001_brain_mask.nii.gz`, you'll have to resample that image to match the voxels in your functional scans.  You can do that, using a technique like the one in `resample_templates.py`.  Let me know if you'd like to do that, but it's entirely optional.

If the hashes are the same then please use whatever download method you like - because the hashes say that the resulting data is identical.  Maybe, to be clever, you could calculate and check in the hashes on the images from the big download, and then use them to check the images you pull down one by one?

**Comment by soazig** [comment]
Hello @matthew-brett 

Thank you for your answers.
I will stick to the methods used in class for the mask of the normal data.
For the downloaded, I actually created a script to generate the hashes of each file of one directory. I used it in a directory with the same file downloaded with the 2 methods. The hashes are the same.
So at this point, I was going for the one by one download, I still have to discuss this with my group.

The .zip file is very big (14GB), each image file is about 300MB. I don't think such a big download is justified if we can do one by one. But I also don't understand such difference in data size.

Thank you, 
Soazig
